src/models/StuntSheetDot.ts

```diff
@@ -1,18 +1,29 @@
 import { FlowBeat } from "./util/types";
 import Serializable from "./util/Serializable";
 
+// Global ID counter for the next dot
+let NEXT_DOT_ID = 0;
+
 /**
  * Defines the position and direction of a marcher for a specific StuntSheet.
  *
+ * @property id            - Uniquely identifies a dot. Can be -1 if not
+ *                           relevant, e.g. for grapher tool dots. Used for
+ *                           connecting dots between stuntsheets, and labels.
  * @property x             - EW position of the first beat
  * @property y             - NS position of the first beat
- * @property dotLabelIndex - Which label to use in Show.dotLabels
+ * @property dotLabelIndex - Which label to use in Show.dotLabels. This is
+ *                           a cached value for stuntsheets after the first.
  * @property dotTypeIndex  - Which set of continuities to use in
  *                           StuntSheet.dotTypes
+ * @property nextDotId     - The dot in the next stuntsheet to generate
+ *                           a flow toward
  * @property cachedFlow    - Cached so that the flow does not need to be
  *                           recalculated again
  */
 export default class StuntSheetDot extends Serializable<StuntSheetDot> {
+  id!: number;
```

Comment:
Would we rather have a sentinel value of -1 rather than making this nullable?

Reply:
The `!` makes sure that a property is non-null. But I think it makes sense to give it a default value of -1! https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

Reply:
Got it; I saw that you had chosen to use -1 to mean 'ignore this field' instead of 'null' or 'undefined' -- is there a good reason to choose -1 instead of null/undefined for that? I'm OK with keeping as-is, but figured I'd still pose the question.